src/tuneflow_py/models/lyric.py

```diff
@@ -0,0 +1,503 @@
+        del self._proto.words[:]
+        for i, word in enumerate(tokens):
+            start = start_tick + i * tick_per_word
+            # Adjust the last word to match the end tick
+            end = start_tick + (i + 1) * tick_per_word if i < len(tokens) - 1 else end_tick
+            self._proto.words.add(word=word, start_tick=start, end_tick=end)
+        self.sort_words()
+
+    def set_words_from_string(
+        self,
+        input: str,
+        start_tick: int | None = None,
+        end_tick: int | None = None,
+        tokenizer: Optional[LyricTokenizer] = None,
+    ):
+        '''
+        Update lyric word objects of the line from a string of words
+        The string is split into words with a tokenizer if provided
+        Otherwise, the default tokenizer is used
+
+        Args:
+            input (str): The string of words
+            start_tick (int | None): The start tick of the line. Use the original tick if set None
+            end_tick (int | None): The end tick of the line. Use the original tick if set None
+            tokenizer (Optional[LyricTokenizer]): The tokenizer to use
+        '''
+        start_tick = self.get_start_tick() if start_tick is None else start_tick
+        end_tick = self.get_end_tick() if end_tick is None else end_tick
+        tokens = (
+            tokenizer(input) if tokenizer
+            else LyricLine.default_lyric_tokenizer(input)
+        )
+        if len(tokens) == 0:
+            self.clear()
+        self._set_words_from_tokens(tokens, start_tick, end_tick)
+
+    @staticmethod
+    def _get_index_of_word_at_tick(line: song_pb2.LyricLine, tick: int):
+        '''
+        Internal generator implementation that locates word at a given tick.
+        '''
+        if len(line.words) == 0 or tick < LyricLine._get_start_tick(line) or tick >= LyricLine._get_end_tick(line):
+            return -1
+        index = lower_equal(
+            line.words,
+            tick,
+            key=lambda item: item if isinstance(item, int) else item.start_tick
+        )
+        if index in range(len(line.words)):
+            return index
+        return -1
+
+    def get_index_of_word_at_tick(self, tick: int):
+        '''
+        Return the index of the word in this proto that contains the given tick,
+        or -1 if no such word exists.
+
+        Args:
+            tick (int): The tick to search for.
+
+        Return:
+            int: The index of the line that contains the tick, or -1 if not found.
+        '''
+        return LyricLine._get_index_of_word_at_tick(self._proto, tick)
+
+    def create_word(self, word: str, start_tick: int, end_tick: int, resolve_order: bool = True):
+        if self.is_empty():
+            # Remove the placeholder
+            del self._proto.words[:]
+        proto = self._proto.words.add(word=word, start_tick=start_tick, end_tick=end_tick)
+        if resolve_order:
+            self.sort_words()
+        return LyricWord(
```

Comment:
I think the principle here is that you don't need to provide more params if you already provided the proto and the parent